cli.js

```diff
@@ -112,17 +115,19 @@ api.on('error', logger.unhandledError);
 api.on('stdout', logger.stdout);
 api.on('stderr', logger.stderr);
 
-api.run()
-	.then(function () {
-		logger.finish();
-		logger.exit(api.failCount > 0 || api.rejectionCount > 0 || api.exceptionCount > 0 ? 1 : 0);
-	})
-	.catch(function (err) {
-		if (err.name === 'AvaError') {
-			console.log('  ' + colors.error(figures.cross) + ' ' + err.message);
-		} else {
-			console.error(colors.stack(err.stack));
-		}
-
-		logger.exit(1);
-	});
+if (cli.flags.watch) {
+	watcher.start(logger, api);
+} else {
+	api.run()
+		.then(function () {
+			logger.finish();
+			logger.exit(api.failCount > 0 || api.rejectionCount > 0 || api.exceptionCount > 0 ? 1 : 0);
+		})
+		.catch(function (err) {
+			// Don't swallow exceptions. Note that any expected error should already
+			// have been logged.
+			process.nextTick(function () {
```

Comment:
Also, `nextTick` is [deprecated since `0.10.0`](https://nodejs.org/en/blog/release/v0.10.0/#faster-process-nexttick)